The test catching the panic  seems to work in principle but fails with a strange typecasting error:
https://github.com/gontract/gontract/actions/runs/19864014190/job/56921528435?pr=7 :
```console
...
Run make test
ok  	github.com/gontract/gontract	0.003s
?   	github.com/gontract/gontract/cmd/example_sqrt_fail_post	[no test files]
?   	github.com/gontract/gontract/cmd/example_sqrt_fail_pre	[no test files]
recovered from panic(CalculateSqrt):  'precondition not satisfied (square root undefined for negative numbers) - software bug!?'--- FAIL: TestCalculateSqrt (0.00s)
    main_test.go:38: 
        	Error Trace:	/home/runner/work/gontract/gontract/cmd/example_sqrt_success/main_test.go:38
        	Error:      	Not equal: 
        	            	expected: float64(-1)
        	            	actual  : int(-1)
        	Test:       	TestCalculateSqrt
FAIL
```

I don't see yet how `int(-1)` can be returned there...

--- a/cmd/example_sqrt_success/main_test.go
+++ b/cmd/example_sqrt_success/main_test.go
@@ -13,7 +13,7 @@ func CheckCalculateSqrt(num float64) (ret float64) {
 
 	defer func() {
 		if r := recover(); r != nil {
-			fmt.Printf("CalculateSqrt paniced: '%s'", r.(string))
+			fmt.Printf("recovered from panic(CalculateSqrt):  '%s'", r.(string))
 			ret = -1
 		}
 	}()
@@ -31,4 +31,9 @@ func TestCalculateSqrt(t *testing.T) {
 	num = 4
 	root = CheckCalculateSqrt(num)
 	assert.Equal(t, num, root*root)
+
+	num = -1
+	root = CheckCalculateSqrt(num)
+	// on panic, -1 is returned:
+	assert.Equal(t, root, -1.0)
 }

